pkg/nuctl/command/deploy.go

```diff
@@ -160,9 +161,10 @@ func newDeployCommandeer(rootCommandeer *RootCommandeer) *deployCommandeer {
 
 			commandeer.rootCommandeer.loggerInstance.DebugWith("Deploying function", "functionConfig", commandeer.functionConfig)
 			_, deployErr := rootCommandeer.platform.CreateFunction(&platform.CreateFunctionOptions{
```

Comment:
```golang
enriched, err := rootCommandeer.platform.EnrichFunctionConfig(commandeer.functionConfig)
commandeer.overrideHTTPTrigger(enriched, commandeer.overrideHTTPTriggerServiceType)
_, deployErr := rootCommandeer.platform.CreateFunction(&platform.CreateFunctionOptions{
```

To override enriched values, you may use platform's _EnrichFunctionConfig()_ function. that way, you would save yourself the trouble of adding special attributes on `CreateFunctionOptions` which open a lead to following scenarios where you need to override values after it gets enriched. I prefer to keep nuctl overriding logics on nuctl, the same it is and not push it down the hole to platform implementation (i.e.: calling `overrideHTTPTriggerServiceType` from within platform to fulfil nuctl flags)

I know the pitfall for my suggestion is that enrichment may happen twice, but that is an issue can be resolved by adding enrichment and validation function calls outside of the create function, or, in other words, it is the callee issue and can be resolved easily.